pysat/instruments/pysat_netcdf.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# Full license can be found in License.md
+# Full author list can be found in .zenodo.json file
+# DOI:10.5281/zenodo.1199703
+# ----------------------------------------------------------------------------
+"""General Instrument for loading pysat-written netCDF files.
+
+Properties
+----------
+platform
+    'pysat', will be updated if file contains a platform attribute
+name
+    'netcdf', will be updated if file contains a name attribute
+tag
+   '', will be updated if file contains a tag attribute
+inst_id
+   '', will be updated if file contains an inst_id attribute
+
+Note
+----
+Only tested against pysat created netCDF files
+
+Examples
+--------
+::
+
+    import pysat
+
+    # Load a test Instrument
+    inst = pysat.Instrument("pysat", "testing")
+    inst.load(date=inst.inst_module._test_dates[''][''])
+
+    # Create a NetCDF file
+    fname = "test_pysat_file_%Y%j.nc"
+    inst.to_netcdf4(fname=inst.date.strftime(fname))
+
+    # Load the NetCDF file
+    file_inst = pysat.Instrument(
+        "pysat", "netcdf", temporary_file_list=True, directory_format="./",
+        file_format="test_pysat_file_{year:04}{day:03}.nc")
+    file_inst.load(date=inst.inst_module._test_dates[''][''])
+
+"""
+
+import datetime as dt
+import numpy as np
+import warnings
+
+import pysat
+
+logger = pysat.logger
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+platform = 'pysat'
+name = 'netcdf'
+tags = {'': ''}
+inst_ids = {'': tag for tag in tags.keys()}
+
+# ----------------------------------------------------------------------------
+# Instrument testing attributes
+
+_test_dates = {'': {'': dt.datetime(2009, 1, 1)}}
+_test_download = {'': {'': False}}
+_test_download_ci = {'': {'': False}}
+
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+def init(self, pandas_format=True):
+    """Initialize the Instrument object with instrument specific values."""
+
+    self.acknowledgements = "Acknowledgements missing from file"
+    self.references = "References missing from file"
+    self.pandas_format = pandas_format
+
+    return
+
+
+def clean(self):
+    """Clean the file data."""
+    return
+
+
+def preprocess(self):
+    """Extract Instrument attrs from file attrs loaded to `Meta.header`."""
+
+    if hasattr(self.meta, "header"):
+        for iattr in ['platform', 'name', 'tag', 'inst_id', 'acknowledgements',
+                      'references']:
+            if hasattr(self.meta.header, iattr):
+                setattr(self, iattr, getattr(self.meta.header, iattr))
+
+    return
+
+
+# ----------------------------------------------------------------------------
+# Instrument functions
+def list_files(tag='', inst_id='', data_path=None, format_str=None):
+    """Produce a list of pysat-written NetCDF files.
```

Comment:
Can we use the generalized `methods.general.list_files` here?

Reply:
Yup!